examples/lua/lua.c

```diff
@@ -0,0 +1,658 @@
+/*
```

Comment:
As far as I see, a large portion of this file is internal code. This should be in pkg/lua/contrib

Reply:
A lot of this is implementation of a cli program, rather than the api - similar to e.g. examples/gcoap/gcoap_cli.c. So, is it best to think now about the API which is going to be exposed to an application and rearrange/separate out the lua.c file accordingly? My approach was to leave it to "evolve", but maybe the more usual approach is that code merged to master needs to be well thought out everywhere, is this true? If so will spend some time here...

Reply:
Further thoughts: just spoke to @miri64 who suggested I think more about what the user would need to know about to implement a program (goes in example) and what they don't need to know (goes in pkg)

Reply:
I'm working on Lua now. The lua.c file is going away. It contains a REPL that is best implemented in Lua itself.